File: flutter_sdk/two_factor_auth_sdk/lib/src/models/activation_request.dart
/// Activation request model
class ActivationRequest {
  final String deviceId;
  final String otpCode;
  final String? username;
  final String? issuer;

  ActivationRequest({
    required this.deviceId,
    required this.otpCode,
    this.username,
    this.issuer,
  });

  Map<String, dynamic> toJson() {
    return {
      'deviceId': deviceId,
      'otpCode': otpCode,
      if (username != null) 'username': username,
      if (issuer != null) 'issuer': issuer,
    };
  }

  factory ActivationRequest.fromJson(Map<String, dynamic> json) {
    return ActivationRequest(
      deviceId: json['deviceId'] as String,
      otpCode: json['otpCode'] as String,
      username: json['username'] as String?,
      issuer: json['issuer'] as String?,
    );
  }
}

File: NetCore/TwoFactorAuth.Core/Models/ActivationRequest.cs
// SPDX-FileCopyrightText: (C) 2025 NetKnights GmbH <https://netknights.it>
// SPDX-License-Identifier: AGPL-3.0-or-later

namespace TwoFactorAuth.Core.Models;

/// <summary>
/// Device activation request model
/// </summary>
public class ActivationRequest
{
    public string DeviceId { get; set; } = string.Empty;
    public string? DeviceName { get; set; }
    public string? Platform { get; set; }
    public string? OsVersion { get; set; }
    public string? Model { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Issuer { get; set; } = string.Empty;
}
